tests/Kinepolis/Trailer/YoutubeTrailerRepositoryTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Kinepolis\Trailer;
+
+use Broadway\UuidGenerator\UuidGeneratorInterface;
+use CultuurNet\UDB3\Json;
+use CultuurNet\UDB3\Model\ValueObject\MediaObject\Video;
+use CultuurNet\UDB3\Model\ValueObject\Translation\Language;
+use CultuurNet\UDB3\Model\ValueObject\Web\Url;
+use CultuurNet\UDB3\SampleFiles;
+use Google\Service\YouTube\Resource\Search;
+use Google_Service_YouTube;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+final class YoutubeTrailerRepositoryTest extends TestCase
+{
+    private TrailerRepository $trailerRepository;
+
+    /**
+     * @var Google_Service_YouTube&MockObject
+     */
+    private $youtubeClient;
+
+    private string $channelId;
+
+    /**
+     * @var UuidGeneratorInterface&MockObject
+     */
+    private $uuidGenerator;
+
+    /**
+     * @var Search&MockObject
+     */
+    private $search;
+
+    public function setUp(): void
+    {
+        $this->search = $this->createMock(Search::class);
+        $this->youtubeClient = $this->createMock(Google_Service_YouTube::class);
+        $this->youtubeClient->search = $this->search;
+        $this->channelId = 'mockChannelId';
+        $this->uuidGenerator = $this->createMock(UuidGeneratorInterface::class);
+
+        $this->trailerRepository = new YoutubeTrailerRepository(
+            $this->youtubeClient,
+            $this->channelId,
+            $this->uuidGenerator,
+            true
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_will_only_search_when_enabled(): void
+    {
+        $disabledTrailerRepository = new YoutubeTrailerRepository(
+            $this->youtubeClient,
+            $this->channelId,
+            $this->uuidGenerator,
+            false
```

Comment:
Now that I see this code it's a bit weird that the repo can be disabled "internally". I think it makes more sense to put the disabling on another higher level and never call the any methods of the repo. But no need to change it in this pull request.

Reply:
Yeah, but the original idea for this to be a temporary flag, to quickly disable the feature if we got weird results. Now that it works, I think it can safely removed during: https://jira.publiq.be/browse/III-6184 but I'll have to wait till Kristof gets back before I pickup that ticket, because I do not know how the deployment pipeline will handle removing the files in appconfig.

Reply:
I guess we could already do the code part without the help of Kristof? The other parts of the ticket can be discussed during a backend meeting?